src/action/action_built.cpp

```diff
@@ -325,8 +325,8 @@ static std::size_t FindCFramePercent(const std::vector<CConstructionFrame> &cfra
 {
 	const auto it =
 		ranges::find_if(cframes, [&](const auto &frame) { return percent < frame.Percent; });
```

Comment:
I played with unit-testing this spot, but it got quite large. Not sure if we should add it.

Reply:
I didn't look a lot for UTs here :/
Responsability split between Construction and action...

Reply:
I will clean the test up and create a PR (maybe today).